### ISSUE TITLE
Existing images not regenerated when image URL is updated

### ISSUE THREAD

**Issue by wfdd** [everypolitician/everypolitician#441]
See, e.g., Serdar Denktaş in http://everypolitician.org/northern-cyprus/assembly/term-table/14.html or Nivi Olsen in http://everypolitician.org/greenland/inatsisartut/term-table/12.html.

**Comment by wfdd** [comment]
This is coming up again and again as I play Gender Balance.

**Comment by wfdd** [comment]
[This](https://github.com/everypolitician/image_cache/blob/f40d0b6a921efd02bc37ea4c491121cca5eea125/lib/image_cache/cacher.rb#L46) appears to be the offending line.  You could save the image URL along with the person ID in the index and read from that, which would work most™ of the time.

**Comment by tmtmtmtm** [comment]
Yep. We can't just removing that line, of course, as it would cause each of the images to be re-spidered dozens of times per day, which is a bit too aggressive (there are several countries where I'm fairly sure their site would fall over if we did that, and a few where there'd be a high risk of getting blacklisted). 

As you suggest, I think the best approach here would be for the cache to keep track of source URLs, rather than just keying it by the politician ID. We'd ideally want to do that in a way that allowed for multiple images per person, too, as keeping track of images from different sources has been on the TODO for quite a while now. 

In the meantime, though, we can always trigger a full refresh for any country by hand. I've done it for the ones listed above, and quite a few others — just drop a note here if there are others you'd like us to rebuild.

**Comment by wfdd** [comment]
> just drop a note here if there are others you'd like us to rebuild 

Off the top of my head, Angola and Guyana; and also Singapore after everypolitician/everypolitician-data#12533 is merged - it has a new photo of Azmoon Ahmad.  Any parliament that uses placeholder images would be a prime candidate for updating.
